veil-macros/src/lib.rs

```diff
@@ -1,87 +1,31 @@
+//! Macros for `veil`
+
+#![deny(missing_docs)]
+#![deny(rustdoc::broken_intra_doc_links)]
```

Comment:
This is not checked in CI, we should add
`cargo doc --document-private-items --workspace --all-features --no-deps`
as a step(copied from prima_datadog.rs)

Reply:
Good catch, I'll do that in a separate PR. PLATFORM-867